packages/calcite-components/src/components/menu-item/menu-item.tsx

```diff
@@ -266,61 +266,54 @@ export class CalciteMenuItem implements LoadableComponent, T9nComponent, Localiz
   };
 
   private keyDownHandler = async (event: KeyboardEvent): Promise<void> => {
-    // opening and closing of submenu is handled here. Any other functionality is bubbled to parent menu.
-    if (event.key === " " || event.key === "Enter") {
-      this.selectMenuItem(event);
-      if (
-        this.hasSubmenu &&
-        (!this.href || (this.href && event.target === this.dropdownActionEl))
-      ) {
-        this.open = !this.open;
+    const { hasSubmenu, href, layout, open, submenuItems } = this;
+    const key = event.key;
+    const targetIsDropdown = event.target === this.dropdownActionEl;
+    if (key === " " || key === "Enter") {
+      if (hasSubmenu && (!href || (href && targetIsDropdown))) {
+        this.open = !open;
       }
-      event.preventDefault();
-    } else if (event.key === "Escape") {
-      if (this.open) {
+      if (!(href && targetIsDropdown) && key !== "Enter") {
+        this.selectMenuItem(event);
+      }
+      if (key === " " || (href && targetIsDropdown)) {
+        event.preventDefault();
+      }
+    } else if (key === "Escape") {
+      if (open) {
         this.open = false;
         return;
       }
       this.calciteInternalMenuItemKeyEvent.emit({ event });
       event.preventDefault();
-    } else if (event.key === "ArrowDown" || event.key === "ArrowUp") {
+    } else if (key === "ArrowDown" || key === "ArrowUp") {
       event.preventDefault();
-      if (
-        (event.target === this.dropdownActionEl || !this.href) &&
-        this.hasSubmenu &&
-        !this.open &&
-        this.layout === "horizontal"
-      ) {
+      if ((targetIsDropdown || !href) && hasSubmenu && !open && layout === "horizontal") {
```

Comment:
Can we store `(targetIsDropdown || !href) && hasSubmenu && !open` in a local var and reuse?